app/assets/javascripts/controllers/ems_common/ems_common_form_controller.js

```diff
@@ -298,5 +325,161 @@ ManageIQ.angular.app.controller('emsCommonFormController', ['$http', '$scope', '
+      };
+    } else if (prefix === "amqp") {
+      if ($scope.newRecord) {
+        var amqp_password = $scope.emsCommonModel.amqp_password;
+        var amqp_verify = $scope.emsCommonModel.amqp_verify;
+      } else {
+        var amqp_password = $scope.emsCommonModel.amqp_password === "" ? "" : miqService.storedPasswordPlaceholder;
+        var amqp_verify = $scope.emsCommonModel.amqp_verify === "" ? "" : miqService.storedPasswordPlaceholder;
+      }
+      $scope.postValidationModel.amqp = {
+        amqp_hostname:             $scope.emsCommonModel.amqp_hostname,
+        amqp_api_port:             $scope.emsCommonModel.amqp_api_port,
+        amqp_security_protocol:    $scope.emsCommonModel.amqp_security_protocol,
+        amqp_userid:               $scope.emsCommonModel.amqp_userid,
+        amqp_password:             amqp_password,
+        amqp_verify:               amqp_verify,
+      };
+    } else if (prefix === "metrics") {
+      if ($scope.newRecord) {
+        var metrics_password = $scope.emsCommonModel.metrics_password;
+        var metrics_verify = $scope.emsCommonModel.metrics_verify;
+      } else {
+        var metrics_password = $scope.emsCommonModel.metrics_password === "" ? "" : miqService.storedPasswordPlaceholder;
+        var metrics_verify = $scope.emsCommonModel.metrics_verify === "" ? "" : miqService.storedPasswordPlaceholder;
+      }
+      var metricsValidationModel = {
+        metrics_hostname:          $scope.emsCommonModel.metrics_hostname,
+        metrics_api_port:          $scope.emsCommonModel.metrics_api_port,
+        metrics_userid:            $scope.emsCommonModel.metrics_userid,
+        metrics_password:          metrics_password,
+        metrics_verify:            metrics_verify,
+      };
+      $scope.postValidationModel['metrics'] = metricsValidationModel;
+    } else if (prefix === "ssh_keypair") {
+      if ($scope.newRecord) {
+        var ssh_keypair_password = $scope.emsCommonModel.ssh_keypair_password;
+      } else {
+        var ssh_keypair_password = $scope.emsCommonModel.ssh_keypair_password === "" ? "" : miqService.storedPasswordPlaceholder;
+      }
+      $scope.postValidationModel['ssh_keypair'] = {
+        ssh_keypair_userid:        $scope.emsCommonModel.ssh_keypair_userid,
+        ssh_keypair_password:      ssh_keypair_password,
+      }
+    } else if (prefix === "service_account") {
+      $scope.postValidationModel['service_account'] = {
+        service_account:           $scope.emsCommonModel.service_account
+      }
+    } else if (prefix === "hawkular") {
+      $scope.postValidationModel['hawkular'] = {
+        hawkular_hostname:         $scope.emsCommonModel.hawkular_hostname,
+        hawkular_api_port:         $scope.emsCommonModel.hawkular_api_port,
+      }
+    }
+  };
+
+  $scope.validateClicked = function($event, authType, url, formSubmit) {
+    $scope.authType = authType;
+    miqService.validateWithREST($event, authType, url, formSubmit)
```

Comment:
potential bug with `formSubmit` .. you're just passing whatever you got as a param, but this function only works if `formSubmit == true` - else `validateWithREST` returns undefined and no `.then` on undefined :)

And another thing, if you wrap the `miqRESTAjaxButton` value in `$q.when(..)` (in miq_service), then you don't need that `$scope.$apply` here..